app/assets/javascripts/govuk_publishing_components/components/contextual-guidance.js

```diff
@@ -15,7 +15,7 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
   ContextualGuidance.prototype.handleFocus = function (event) {
     this.hideAllGuidance()
-    if (!event.target.dataset.contextualGuidanceHideOnly) {
+    if (!event.target.getAttribute('contextualGuidanceHideOnly')) {
```

Comment:
Ooh, yes, potentially. Good spot. I'll check.

Reply:
I can't see where this attribute appears - there doesn't seem to be a test for it, and the only instance I can find of `contextualGuidanceHideOnly` is in this file. @alex-ju is this something in the publishing applications? Is it needed?

Reply:
so, the equivalent of `dataset.contextualGuidanceHideOnly` is getAttribute('data-contextual-guidance-hide-only')` (attributes get camelcased when added to the dataset object). it used to be a feature in `content-publisher`, but must have ben removed as I can't see any instance of it atm.

Reply:
Thanks both. I've updated the code and added a test for this. Although I'm uncertain if I understand this functionality - it seems like this option allows us to hide the contextual guidance for certain elements, so that they never appear again? Why is that useful? Or have I misunderstood?

Reply:
It was more of an exception to the rule. So the contextual guidance was designed to be visible when a certain input element is focused, and be hidden when the input is 'blurred'. This exception was set in place for a certain case where we only wanted to make sure the contextual guidance is hidden on blur, without showing it on focus. The behaviour was documented in the commits in `content-publisher`, before the component got moved to the library – I guess one of the downsides of developing a component at the app-level.

As the feature is not being used anymore, I don't see value in keeping it around – arguably not too relevant to this PR.

Reply:
Great. I'll raise a separate PR once this is merged to take it out, thanks for the context.